### PR TITLE
Expand list of allowed licenses to include BSD

### PR DESCRIPTION
**Pull request by academo**
- Expand allowed licences and use regex instead of strings
- Add clarification comment

### CODE REVIEW

**Review by sympatheticmoose — CHANGES_REQUESTED** [review]
The readme section on the analyzers lists the previously accepted list. Given this expansion I suggest adding under https://github.com/grafana/plugin-validator#license a new statement:

> We accept the following licenses for community plugins: BSD, MIT, Apache 2.0, LGPL3, GPL3, AGPL3.

Then change the description for the license analyzer in https://github.com/grafana/plugin-validator#analyzers to be:

```
Ensures the [license type](https://github.com/grafana/plugin-validator#license) specified is allowed.
```
Eventually we should also update the main docs as well to make sure this is clear up front.

File: README.md [hunk 1]
```diff
@@ -89,13 +89,14 @@ Usage plugincheck2:
         If set, plugincheck returns non-zero exit code for warnings
```
Comment (academo) [inline README.md:89]: I applied prettier formatting to this file for more legibility

**Review by sympatheticmoose — APPROVED** [review]
Readme LTGM - thank you! May still want to update the error message in `license.go` but thats non-blocking

**Review by tolzhabayev — APPROVED** [review]
LGTM

@sympatheticmoose if you are up to it and want to dig into how our main docs at grafana/grafana work - you could add a "license" section to https://grafana.com/docs/grafana/latest/developers/plugins/publishing-and-signing-criteria/

File: pkg/analysis/passes/license/license_test.go [hunk 2]
```diff
@@ -76,13 +76,29 @@ func TestValidMitLicense(t *testing.T) {
...
+		},
+		Report: interceptor.ReportInterceptor(),
+	}
+
+	_, err := Analyzer.Run(pass)
+	require.NoError(t, err)
+	require.Len(t, interceptor.Diagnostics, 0)
+}
+
+func TestInvalidUnilicenseLicense(t *testing.T) {
```
Comment (xnyo) [inline pkg/analysis/passes/license/license_test.go:95]: Nit: Typo: The name of the license is "unlicense" :smiley:

**Pull request merged**